usaspending_api/common/threaded_data_loader.py

```diff
@@ -52,117 +57,129 @@ def __init__(self, model_class, threads=10, field_map={}, value_map={}, collisio
     def load_from_file(self, filepath):
         self.logger.info('Started processing file ' + filepath)
         # Create the Queue object - this will hold all the rows in the CSV
-        row_queue = queue.Queue(maxsize=100)
-        self.exit_flag = False
-
-        # Spawn our threads
+        row_queue = JoinableQueue()
+
+        references = {
+            "collision_field": self.collision_field,
+            "collision_behavior": self.collision_behavior,
+            "logger": self.logger,
+            "pre_row_function": self.pre_row_function,
+            "post_row_function": self.post_row_function,
+            "fields": self.fields.copy()
+        }
+
+        # Spawn our processes
+        db.connections.close_all()
```

Comment:
When you spawn a new process Django copies the current database connection if it exists for that process to use. That means if you have 10 processes they're all trying to use the same connection which can lead to some deadlock/loss of performance. By closing all connections before spawning the processes, you ensure each process gets their own connection.